invenio_app_rdm/theme/assets/js/invenio_app_rdm/search/SearchResults/SearchResults.js

```diff
@@ -0,0 +1,92 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2019 CERN.
+// Copyright (C) 2019 Northwestern University.
+//
+// Invenio App RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import { Container, Grid } from "semantic-ui-react";
+import _truncate from "lodash/truncate";
+import {
+  Count,
+  LayoutSwitcher,
+  Pagination,
+  ResultsMultiLayout,
+  ResultsPerPage,
+  Sort,
+  EmptyResults,
+  Error,
+  ResultsLoader,
+  withState
+} from "react-searchkit/esm";
+
+import { ResultsGridItem, ResultsListItem } from "./components";
+import { config } from "../config";
+
+const SpanWithMargin = ({ text, margin }) => {
+  const size = "0.3em";
+  let style;
+  switch (margin) {
+    case "left":
+      style = { marginLeft: size };
+      break;
+    case "right":
+      style = { marginRight: size };
+      break;
+    default:
+      style = { margin: `0 ${size}` };
+  }
+  return <span style={style}>{text}</span>;
+};
+
+class Results extends Component {
+  render() {
+    const { total } = this.props.currentResultsState.data;
```

Comment:
question: `this.props.currentResultsState` comes from `withState`? rather than using some form of inheritance? Just curious about choice.

Reply:
Yes `withState` is a high order function that gives you the chance to hook in searchkit's internal store and consume it in your component. Is defined [here](https://github.com/inveniosoftware/react-searchkit/blob/master/src/lib/components/HOC/withState.js)!

Reply:
Haha, I am going to need to learn me some more react, 'cause that function is gibberish to me right now :smile: